tools/kvmexit.py

```diff
@@ -163,90 +164,182 @@ def valid_args_list(args):
-    "MSR_READ",
-    "MSR_WRITE",
-    "INVALID_STATE",
-    "MSR_LOAD_FAIL",
-    "N/A",
-    "MWAIT_INSTRUCTION",
-    "MONITOR_TRAP_FLAG",
-    "N/A",
-    "MONITOR_INSTRUCTION",
-    "PAUSE_INSTRUCTION",
-    "MCE_DURING_VMENTRY",
-    "N/A",
-    "TPR_BELOW_THRESHOLD",
-    "APIC_ACCESS",
-    "EOI_INDUCED",
-    "GDTR_IDTR",
-    "LDTR_TR",
-    "EPT_VIOLATION",
-    "EPT_MISCONFIG",
-    "INVEPT",
-    "RDTSCP",
-    "PREEMPTION_TIMER",
-    "INVVPID",
-    "WBINVD",
-    "XSETBV",
-    "APIC_WRITE",
-    "RDRAND",
-    "INVPCID",
-    "VMFUNC",
-    "ENCLS",
-    "RDSEED",
-    "PML_FULL",
-    "XSAVES",
-    "XRSTORS",
-    "N/A",
-    "N/A",
-    "UMWAIT",
-    "TPAUSE"
+# format setting, hard code in case of no header files
+exit_reasons_Intel = (
+    "EXCEPTION_NMI 0",
+    "EXTERNAL_INTERRUPT 1",
+    "TRIPLE_FAULT 2",
+    "INIT_SIGNAL 3",
+    "INTERRUPT_WINDOW 7",
+    "NMI_WINDOW 8",
+    "TASK_SWITCH 9",
+    "CPUID 10",
+    "HLT 12",
+    "INVD 13",
+    "INVLPG 14",
+    "RDPMC 15",
+    "RDTSC 16",
+    "VMCALL 18",
+    "VMCLEAR 19",
+    "VMLAUNCH 20",
+    "VMPTRLD 21",
+    "VMPTRST 22",
+    "VMREAD 23",
+    "VMRESUME 24",
+    "VMWRITE 25",
+    "VMOFF 26",
+    "VMON 27",
+    "CR_ACCESS 28",
+    "DR_ACCESS 29",
+    "IO_INSTRUCTION 30",
+    "MSR_READ 31",
+    "MSR_WRITE 32",
+    "INVALID_STATE 33",
+    "MSR_LOAD_FAIL 34",
+    "MWAIT_INSTRUCTION 36",
+    "MONITOR_TRAP_FLAG 37",
+    "MONITOR_INSTRUCTION 39",
+    "PAUSE_INSTRUCTION 40",
+    "MCE_DURING_VMENTRY 41",
+    "TPR_BELOW_THRESHOLD 43",
+    "APIC_ACCESS 44",
+    "EOI_INDUCED 45",
+    "GDTR_IDTR 46",
+    "LDTR_TR 47",
+    "EPT_VIOLATION 48",
+    "EPT_MISCONFIG 49",
+    "INVEPT 50",
+    "RDTSCP 51",
+    "PREEMPTION_TIMER 52",
+    "INVVPID 53",
+    "WBINVD 54",
+    "XSETBV 55",
+    "APIC_WRITE 56",
+    "RDRAND 57",
+    "INVPCID 58",
+    "VMFUNC 59",
+    "ENCLS 60",
+    "RDSEED 61",
+    "PML_FULL 62",
+    "XSAVES 63",
+    "XRSTORS 64",
+    "UMWAIT 67",
+    "TPAUSE 68"
+)
+
+exit_reasons_AMD = (
+    "READ_CR0 0",
+    "READ_CR2 2",
+    "READ_CR3 3",
+    "READ_CR4 4",
+    "READ_CR8 8",
+    "WRITE_CR0 16",
+    "WRITE_CR2 18",
+    "WRITE_CR3 19",
+    "WRITE_CR4 20",
+    "WRITE_CR8 24",
+    "READ_DR0 32",
+    "READ_DR1 33",
+    "READ_DR2 34",
+    "READ_DR3 35",
+    "READ_DR4 36",
+    "READ_DR5 37",
+    "READ_DR6 38",
+    "READ_DR7 39",
+    "WRITE_DR0 48",
+    "WRITE_DR1 49",
+    "WRITE_DR2 50",
+    "WRITE_DR3 51",
+    "WRITE_DR4 52",
+    "WRITE_DR5 53",
+    "WRITE_DR6 54",
+    "WRITE_DR7 55",
+    "EXCP_BASE_DE 64",
+    "EXCP_BASE_DB 65",
+    "EXCP_BASE_BP 67",
+    "EXCP_BASE_OF 68",
+    "EXCP_BASE_BR 69",
+    "EXCP_BASE_UD 70",
+    "EXCP_BASE_NM 71",
+    "EXCP_BASE_DF 72",
+    "EXCP_BASE_TS 74",
+    "EXCP_BASE_NP 75",
+    "EXCP_BASE_SS 76",
+    "EXCP_BASE_GP 77",
+    "EXCP_BASE_PF 78",
+    "EXCP_BASE_MF 80",
+    "EXCP_BASE_AC 81",
+    "EXCP_BASE_MC 82",
+    "EXCP_BASE_XM 83",
```

Comment:
This does NOT match the kernel header file https://github.com/torvalds/linux/blob/master/arch/x86/include/uapi/asm/svm.h
```c
#define SVM_EXIT_EXCP_BASE     0x040
#define SVM_EXIT_LAST_EXCP     0x05f
```

Reply:
There DO have some specific exit reasons based on SVM_EXIT_EXCP_BASE:

https://github.com/torvalds/linux/blob/master/arch/x86/include/uapi/asm/kvm.h#L17
https://github.com/torvalds/linux/blob/master/tools/arch/x86/include/uapi/asm/svm.h#L145